_posts/2018-05-24-Spring-IO-2018.md

```diff
@@ -0,0 +1,777 @@
+You define the sample size of the request to trace yourself as you don't want to trace everything especially if you have a high load.
+This sample size is configurable at runtime, especially handy to debug errors in production.
+[Zipkin](https://zipkin.io){:target="_blank"} with its UI helps you to see the timing information visually and is a good tracing backend for Spring applications.
+
+Using Spring Cloud Sleuth, the tracing instrumentation via [Zipkin's Brave](https://github.com/openzipkin/brave){:target="_blank"} is auto-configured.
+Via properties you can configure things such as the sampling probability and whether certain endpoints should to be skipped.
+It is also compatible with the [OpenTracing](http://opentracing.io/){:target="_blank"} standard that is being developed under the wings of the [CNCF](https://www.cncf.io/){:target="_blank"}.
+
+### Correlation everywhere
+
+Having set up all of these, you now have correlated logging, metrics and tracing across your system, and you can find the data from each based on identifiers.
+
+### Observability cycle
+
+If an issue produces itself we can take the following steps to troubleshoot and bandage the situation:
+
+* The issue should have been reported via an alert or report
+* We check the metrics of our system
+* If needed, we check the tracing data
+* If needed, we check the logs
+* Based on the gathered information we can triage the issue and make adjustments to prevent a recurrence
+
+### Key takeaways
+
+System wide observability is crucial in distributed architectures.
+The tools to help you with this exist and Spring makes it easy to integrate them in your system as the most common cases are covered out-of-the-box or easily configurable.
+Use the right tool for the job and synergise across the different tools.
+
+## Day 2: Talks & Workshops
+
+## Machine learning exposed: The fundamentals by [James Weaver](https://twitter.com/JavaFXpert){:target="_blank"}
+
+<span class="image left"><img class="p-image" alt="James Weaver" src="/img/spring-io-2018/james-weaver.jpg"></span>
+
+Machine Learning is a hot topic in tech land with all kinds of applications like predicting property prices, forecasting weather, self driving cars, plants classification and so on. James gave a brief overview about the fundamentals of Machine Learning and its applications.
+
+But how can we define Machine Learning? [Andrew Ng](https://twitter.com/AndrewYNg){:target="_blank"}, Co-founder of Coursera and Adjunct Professor of Stanford University defined Machine learning in his introduction course "Welcome To Machine Learning"  <sup>1</sup> as "*Machine Learning is the science of getting computers to learn, without being explicitly programmed*". An example that Andrew gave was a cleaning robot that can tidy your house. Instead you program the algorithm explicitly on how it should clean. You can for instance let the robot watch you while you demonstrate the tasks on how it should clean and learn from it.
```

Comment:
-> "Instead of programming the algorithm explicitly on how it should clean, you can for instance let the robot watch you....." -> one sentence